app.R

```diff
@@ -150,180 +150,159 @@ shinyApp (
-            rownames(tcga) = tcga$gene
-            tcga <- tcga[,-1]
-            
-            
-            bysubtype <- filter(clinicaldata_tmt, clinicaldata_tmt$Cyto_Risk %in% subtypesToPlot)
-            plottinggene <- tcga[,colnames(tcga) %in% bysubtype$UPN]
-            plottinggene <- t(plottinggene)
-            plottinggene <- as.data.frame(plottinggene)
-            plottinggene$UPN <- rownames(plottinggene)
-            rownames(plottinggene) <- NULL
-            
-            # Data cleanup and preparing for output
-            finalmatrix <- merge(plottinggene, bysubtype, by="UPN")
-            finalmatrix <- finalmatrix[,c(2,3,4,5,7)]
-            colnames(finalmatrix) <- c("Log2.Expression","Name","TCGA_ID","TCGA_Name","Cyto_risk")
-            levels(finalmatrix$Cyto_risk) <- c("Favourable","Healthy Lin-","Intermediate","Adverse")
-            finalmatrix$Cyto_risk <- factor(finalmatrix$Cyto_risk,levels=c("Favourable","Intermediate","Adverse","Healthy Lin-"))
-            finalmatrix$Gene <- genesToPlot
-            finalmatrix <- finalmatrix[,c(2,3,4,5,1,6)]
-            dmtc <<- finalmatrix
-            
-            g <- ggplot(finalmatrix,aes(Cyto_risk, Log2.Expression, text = paste0("UPN ID: ",Name,"<br />TCGA Sample ID: ", TCGA_ID))) + geom_quasirandom(size = 0.8) + theme_bw() +
-              ggtitle(paste0("Log2 Expression for ",genesToPlot)) + 
-              theme(text=element_text(size=12, family="avenir", face="bold"), axis.text=element_text(size=12, family="avenir", face="bold")) +
-              ylab("Log2 Expression") + xlab("")
-            ggplotly(g, tooltip="text")
+          #SELECT Value,UPN FROM Genes WHERE Gene="DNMT3A" AND Type='TMT'; (Replace DNMT3A with all genes selected)
+          query <- paste(sep="","SELECT Value,UPN FROM Genes WHERE Gene='",genesToPlot, "' AND Type='TMT';")
+          tcga <- dbGetQuery(database,query)
+
+
+          #SELECT UPN, Name, TCGA_ID, TCGA_Name, Cyto_Risk FROM TMT_Clinical WHERE (Cyto_Risk='Poor' OR Cyto_Risk='Good' OR Cyto_Risk='Intermediate' OR Cyto_Risk='Healthy Lin-');
+          query <- "SELECT UPN, Name, TCGA_ID, TCGA_Name, Cyto_Risk FROM TMT_Clinical WHERE ("
+          for(g in subtypesToPlot){
+            query <- paste(query,sep="","Cyto_Risk='",g,"'"," OR ")
           }
+          query <- paste(sep="",substring(query,1, nchar(query)-4),");")
+          df <- dbGetQuery(database,query)
+
+
+          df <- merge(df,tcga, by="UPN")
+          df$Gene <- genesToPlot
+          df <- subset(df, select = -c(UPN) ) # For some reason in the original implementation UPN was dropped
+          levels(df$Cyto_Risk) <- c("Favorable","Healthy Lin-","Intermediate","Adverse")
+          df$Cyto_Risk <- factor(df$Cyto_Risk,levels=c("Favorable","Intermediate","Adverse","Healthy Lin-"))
+          df <- df[,c(2,3,4,5,1,6)]
+          dmtc <<- df
```

Comment:
might be good to use something more descriptive than `dmtc` (what does that stand for?)